crates/subspace-networking/src/create.rs

```diff
@@ -30,6 +38,153 @@ use tracing::info;
+    pub max_circuits: usize,
+    pub max_circuits_per_peer: usize,
+    pub max_circuit_duration: Duration,
+    pub max_circuit_bytes: u64,
+    pub circuit_src_rate_limit_per_peer: rate_limiter::GenericRateLimiterConfig,
+    pub circuit_src_rate_limit_per_ip: rate_limiter::GenericRateLimiterConfig,
+}
+
+impl Default for RelayLimitSettings {
+    fn default() -> Self {
+        let default_relay_config = RelayConfig::default();
+
+        Self {
+            max_reservations: default_relay_config.max_reservations,
+            max_reservations_per_peer: default_relay_config.max_circuits_per_peer,
+            reservation_duration: default_relay_config.reservation_duration,
+            max_circuits: default_relay_config.max_circuits,
+            max_circuits_per_peer: default_relay_config.max_circuits_per_peer,
+            max_circuit_duration: default_relay_config.max_circuit_duration,
+            max_circuit_bytes: default_relay_config.max_circuit_bytes,
+
+            // Copied from the  RelayConfig::default() implementation:
+            // For each peer ID one reservation every 2 minutes with up to 30 reservations per hour.
+            reservation_rate_limit_per_peer: rate_limiter::GenericRateLimiterConfig {
+                limit: NonZeroU32::new(30).expect("30 > 0"),
+                interval: Duration::from_secs(60 * 2),
+            },
+            // For each IP address one reservation every minute with up to 60 reservations per hour.
+            reservation_rate_limit_per_ip: rate_limiter::GenericRateLimiterConfig {
+                limit: NonZeroU32::new(60).expect("60 > 0"),
+                interval: Duration::from_secs(60),
+            },
+            // For each source peer ID one circuit every 2 minute with up to 30 circuits per hour.
+            circuit_src_rate_limit_per_peer: rate_limiter::GenericRateLimiterConfig {
+                limit: NonZeroU32::new(30).expect("30 > 0"),
+                interval: Duration::from_secs(60 * 2),
+            },
+            // For each source IP address one circuit every minute with up to 60 circuits per hour.
+            circuit_src_rate_limit_per_ip: rate_limiter::GenericRateLimiterConfig {
+                limit: NonZeroU32::new(60).expect("60 > 0"),
+                interval: Duration::from_secs(60),
+            },
+        }
+    }
+}
+
+impl RelayLimitSettings {
+    /// Converts `RelayLimitSettings` to libp2p `relay::Config`.
+    pub fn to_relay_config(self) -> RelayConfig {
+        let reservation_rate_limiters = vec![
+            rate_limiter::new_per_peer(self.reservation_rate_limit_per_peer),
+            rate_limiter::new_per_ip(self.circuit_src_rate_limit_per_ip),
+        ];
+
+        let circuit_src_rate_limiters = vec![
+            rate_limiter::new_per_peer(self.circuit_src_rate_limit_per_peer),
+            rate_limiter::new_per_ip(self.circuit_src_rate_limit_per_ip),
+        ];
```

Comment:
I don't think we need this. The only restriction must be to only allow local peers (ideally just memory transport or identify them by known PeerId), the rest should have access denied, so we probably don't need rate limit configs like above.